TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -389,6 +389,29 @@ public override bool Visit(DataDefinition dataDefinition)
 
             }
 
+            //Check if Strict Typedef declaration uses Sync clause
+            if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.InsideTypedef) &&
+                dataDefinition.SemanticData.HasFlag(Symbol.Flags.Sync))
+            {
+                //Typedef instruction => check if it's marked Strict
+                if (dataDefinition.SemanticData.Kind == Symbol.Kinds.Typedef && dataDefinition.SemanticData.HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"Cannot declare Type definition {dataDefinition.Name} with Sync clause because it is Strict.");
+                }
+                //Variable inside Typedef => check if parent typedef is marked Strict
+                else if (dataDefinition.SemanticData.NearestParent(Symbol.Kinds.Typedef).HasFlag(Symbol.Flags.Strict))
+                {
+                    DiagnosticUtils.AddError(dataDefinition, $"{dataDefinition.Name} is part of a declaration using Sync clause in Strict Type definition {dataDefinition.ParentTypeDefinition?.Name}.");
+                }
+
+            }
+            //Check if variable of user defined Strict Type is declared or has a parent declared with Sync clause (flag is inherited so no need to iterate through parents)
+            else if (dataDefinition.SemanticData.HasFlag(Symbol.Flags.HasATypedefType) && 
+                dataDefinition.TypeDefinition?.RestrictionLevel == RestrictionLevel.STRICT &&
```

Comment:
Actually we could use SemanticDomain here too : `dataDefinition.TypeDefinition.SemanticData.HasFlag(Strict)`